### PR TITLE
HDDS-11035. Implement initial version of the Helm chart

### PR DESCRIPTION
**Pull request by dnskr**
## What changes were proposed in this pull request?
There is a common practice to provide an official Helm chart for a product to increase its usage and adoption. I propose to create an initial version of Helm chart to be used by the community for effortless deployment of Apache Ozone on Kubernetes.

The following steps should to be done in order to release initial Helm chart version:
- [x] Implement initial version of the Helm chart;
- [ ] Add CI pipeline for the chart (lint, release etc);
- [ ] Update documentation with the chart details;
- [ ] Release initial version.

**Comments, suggestions and feedback are very welcome!**

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-11027

## How was this patch tested?

Tested manually on Minikube
```sh
helm install ozone charts/ozone
```
Status check
```sh
$ kubectl get pod 
NAME               READY   STATUS    RESTARTS   AGE
ozone-datanode-0   1/1     Running   0          2m11s
ozone-datanode-1   1/1     Running   0          2m10s
ozone-datanode-2   1/1     Running   0          2m9s
ozone-om-0         1/1     Running   0          2m11s
ozone-s3g-0        1/1     Running   0          2m11s
ozone-scm-0        1/1     Running   0          2m11s

$ kubectl get service
NAME                      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
kubernetes                ClusterIP   10.96.0.1        <none>        443/TCP    5h59m
ozone-datanode            ClusterIP   10.98.184.181    <none>        9882/TCP   2m21s
ozone-datanode-headless   ClusterIP   None             <none>        9882/TCP   2m21s
ozone-om                  ClusterIP   10.100.191.112   <none>        9874/TCP   2m21s
ozone-om-headless         ClusterIP   None             <none>        9874/TCP   2m21s
ozone-s3g                 ClusterIP   10.100.105.252   <none>        9878/TCP   2m21s
ozone-s3g-headless        ClusterIP   None             <none>        9878/TCP   2m21s
ozone-scm                 ClusterIP   10.107.10.164    <none>        9876/TCP   2m21s
ozone-scm-headless        ClusterIP   None             <none>        9876/TCP   2m21s
```

### CODE REVIEW

File: charts/ozone/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v2
+name: ozone
+description: The official Helm chart for Apache Ozone
+type: application
+version: 0.1.0
+appVersion: "1.4.0"
```
Comment (adoroszlai) [inline charts/ozone/Chart.yaml:22]: `appVersion` should be defined as `"${project.version}"`, which will be replaced by Maven if we add a section similar to:

https://github.com/apache/ozone/blob/b87f977ef235b7ecb2062c0196dcd0a36157d6e2/hadoop-ozone/dist/pom.xml#L155-L163

It's OK to make this change as part of the "Add CI pipeline for the chart" step later.

**Review by adoroszlai — COMMENTED** [review]
Thank you very much @dnskr for the patch.

I suggest moving `charts` into `hadoop-ozone/dist/src/main/`, which fits Ozone's test/release conventions better than adding it directly at the root dir.  We'll still need to tweak the packaging scripts for `charts` to be included in the distribution.

**Comment by dnskr** [comment]
> I suggest moving `charts` into `hadoop-ozone/dist/src/main/`, which fits Ozone's test/release conventions better than adding it directly at the root dir. We'll still need to tweak the packaging scripts for `charts` to be included in the distribution.

@adoroszlai Thanks for such quick review.

If I understand correctly, you suggest moving `charts` into `hadoop-ozone/dist/src/main/` to make the chart a part of Ozone distribution (release). 

My motivation for putting it to the root was to follow a common approach for Helm charts, so I'm curious what you think. Usually Helm chart lives "outside" an application and has its independent release flow, so the chart can be released more or less often than a product. For instance, Apache Airflow usually releases their chart once for each minor release, Apache Superset releases chart more often then application etc.

So, I propose to follow "monorepo" approach, keep the chart separately and have a dedicated release flow. Also, using the root dir is recommended by Helm [documentation](https://helm.sh/docs/howto/chart_releaser_action/#repository-changes):
```
The charts should be placed under /charts directory at the top-level of the directory tree.
```

What do you think, would it work for Apache Ozone repository approach and structure?

File: charts/ozone/Chart.yaml (hunk 1 above)
Comment (adoroszlai) [reply charts/ozone/Chart.yaml:22]: If charts are not intended to be part of the regular release, this is not an issue.

**Comment by adoroszlai** [comment]
Thanks @dnskr for sharing the reasons to put `charts` at the root dir.

If Helm charts are not meant to be part of the regular release, moving them to the `.../dist` directory is not necessary.  However, in that case I think we should create a new, dedicate repo.  Ozone is not monorepo, there are separate repos for the website and various Docker images.

**Comment by dnskr** [comment]
The dedicated repo approach works great. For instance, it is used by [Argo Project](https://github.com/argoproj/argo-helm) and [Presto](https://github.com/prestodb/presto-helm-charts), and it is also suggested in the documentation. So having `ozone-helm-charts` repo or similar is a good idea and probably gives additional flexibility.

**Comment by adoroszlai** [comment]
Thanks again @dnskr for the PR, let's move this to https://github.com/apache/ozone-helm-charts

**Comment by kerneltime** [comment]
> Thanks again @dnskr for the PR, let's move this to https://github.com/apache/ozone-helm-charts

I know the repo is created but can be rename the report to the Ozone-kubernetes which can include helm charts and possibly a Kubernetes operator itself?

**Comment by dnskr** [comment]
> I know the repo is created but can be rename the report to the Ozone-kubernetes which can include helm charts and possibly a Kubernetes operator itself?

Should it be dedicated repo for operator? For instance, `ozone-operator` or similar with helm chart in `ozone-helm-charts` repo.

**Pull request closed without merging**